util/format-decile-generator.js

```diff
@@ -0,0 +1,77 @@
+// Utils
+var isInverseBenchmark = require('./is-inverse-benchmark');
+
+/**
+ * Generator function to create a
+ * function that formats the deciles based on options
+ *
+ * @param {{
+ *  measureName: string,
+ *  qualityId: string,
+ *  submissionMethod: string,
+ *  measureType: string,
+ *  benchmark: string,
+ *  decile1: string?,
+ *  decile2: string?,
+ *  decile3: string?,
+ *  decile4: string?,
+ *  decile5: string?,
+ *  decile6: string?,
+ *  decile7: string?,
+ *  decile8: string?,
+ *  decile9: string,
+ *  decile10: string,
+ *  isToppedOut: string}} record
+ *  @returns {function}
+ */
+var formatDecileGenerator = function formatDecileGenerator(record) {
+  var isInverseMeasure = isInverseBenchmark(record);
+  var top = isInverseMeasure ? 0 : 100;
+
+  /**
+   * Params correspond to the Array.map signature
+   * @param {string?} decileString - from csv
+   * @param {number} index
+   * @param {Array} array
+   * @return {number | null}
+   */
+  return function(decileString, index, array) {
+    var range           =  decileString ? decileString.match(/([0-9]*[.]?[0-9]+)/g) : null;
+    var nextIndex       = index + 1;
+    var nextDecile      = array[index + 1];
+    var nextDecileRange = nextDecile ? nextDecile.match(/([0-9]*[.]?[0-9]+)/g) : null;
+    // if self is undefined and a previous decile is defined, return null.
+    // if self is undefined and no previous decile is defined, but nextDecile is defined, return nextDecile's lowerbound.
+    // if self is undefined, and the nextDecile is undefined, return null.
+    // if self is defined, but nextDecile is not, find the next defined decile above.
+    // if self is defined and self is Decile 10 return the explicit upper bound or top.
```

Comment:
@marimiyachi Yea, I had similar thoughts as I was writing this. I think we might have over optimized. Might be better to just copy over what was given in the xls, but we still have a few issues/decisions:
1) If Decile 1 is "0-1.02" and Decile 2 is "1.03-2.00", what do we do for values that are less than 1.03 but greater than 1.02 (1.02 < x < 1.03 )? 
2) Still need to parse the Deciles that aren't in "n-m" format, e.g. "<= x", "100", and "0". 

Also we should ask, what will be easiest for when we are the ones maintaining the data and spitting out benchmarks? 

Maybe this calls for a 10 minute chat.